oak_marketplace_emotion_recognition/utils/visualization.py

```diff
@@ -0,0 +1,252 @@
+                                          draw_emotion_recognition_one_frame)
+
+X_OFFSET = 5
+Y_OFFSET = 15
+OVERLAY_PATH = "images/overlay.png"
+
+
+def statistics_update(image, ret, emotions_collector):
+    overlay = cv2.imread(OVERLAY_PATH, cv2.IMREAD_UNCHANGED)
+    if ret:
+        class_name = ret[0].emotions[0].class_name
+        emotions_collector.update_bars(class_name)
+    image = cv2.resize(image, (450, 450))
+    vis_result = draw_emotion_recognition_one_frame(image, ret)
+
+    start_x, start_y = np.clip(vis_result.shape[0] - emotions_collector.graphs_size, 0, vis_result.shape[0]), \
+                       np.clip(vis_result.shape[1] - emotions_collector.graphs_size - Y_OFFSET, 0,
+                               vis_result.shape[0])
+    alpha_mask_overlay = overlay[:, :, 3] / 255.0
+    overlay_background = overlay[..., :3]
+    overlay_image_alpha(vis_result, overlay_background, 0, 0, alpha_mask_overlay)
+    for emotion, emotion_graph in emotions_collector.graphs.items():
+        percent = emotions_collector.get_current_emotion_percent(emotion)
+        text = f'{emotion.upper()} - {percent}%'
+        vis_result = add_class_names_and_percents(vis_result, [start_x, int(vis_result.shape[1] - Y_OFFSET / 2)],
+                                                  text)
+        alpha_mask = emotion_graph.alpha_mask
+        overlay_image_alpha(vis_result, emotion_graph.graph, start_x, start_y, alpha_mask)
+        start_x = np.clip(start_x - emotions_collector.graphs_size - X_OFFSET, 0, vis_result.shape[0])
+    return vis_result
+
+
+def add_class_names_and_percents(image, coords, text):
+    img_h, img_w, _ = image.shape
+    scale = min([img_w, img_h]) / NORM_HEIGHT
+    text_size = int(scale * INFO_TEXT_SIZE)
+    text_offset_y = int(scale * TEXT_OFFSET_Y)
+    text_offset_x = int(scale * TEXT_OFFSET_X)
+    coords = coords[0] - int(2 * scale) + text_offset_x, coords[1] + int(2 * scale) - text_offset_y
+    pil_img = Image.fromarray(image)
+    draw = ImageDraw.Draw(pil_img)
+    montserrat = ImageFont.truetype(MONTSERATT_BOLD_TTF_PATH, text_size)
+    draw.text(coords, text, font=montserrat, fill=WHITE_TEXT_COLOR)
+    return np.array(pil_img)
+
+
+def overlay_image_alpha(img, img_overlay, x, y, alpha_mask=None):
+    y1, y2 = max(0, y), min(img.shape[0], y + img_overlay.shape[0])
+    x1, x2 = max(0, x), min(img.shape[1], x + img_overlay.shape[1])
+
+    y1o, y2o = max(0, -y), min(img_overlay.shape[0], img.shape[0] - y)
+    x1o, x2o = max(0, -x), min(img_overlay.shape[1], img.shape[1] - x)
+
+    if y1 >= y2 or x1 >= x2 or y1o >= y2o or x1o >= x2o:
+        return
+
+    img_crop = img[y1:y2, x1:x2]
+    img_overlay_crop = img_overlay[y1o:y2o, x1o:x2o]
+    alpha = alpha_mask[y1o:y2o, x1o:x2o, np.newaxis]
+    alpha_inv = 1.0 - alpha
+    img_crop[:] = alpha * img_overlay_crop + alpha_inv * img_crop
+
+
+class EmotionBarAnalyzer:
```

Comment:
Use EmotionAnalyzer. EmotionBar is used inside this class so it's OK to name it just EmotionAnalyzer to keep it simpler